unit-tests/py/rspy/ykush.py

```diff
@@ -0,0 +1,258 @@
+Yepkit USB Switchable Hub
+
+Pip page:
+https://pypi.org/project/yepkit-pykush/
+Github:
+https://github.com/Yepkit/pykush
+"""
+
+
+from rspy import log
+import time
+
+from rspy import usb_relay
+
+if __name__ == '__main__':
+    import os, sys, getopt
+    def usage():
+        ourname = os.path.basename( sys.argv[0] )
+        print( 'Syntax: ykush [options]' )
+        print( '        Control the YKUSH USB hub' )
+        print( 'Options:' )
+        print( '        --enable       Enable all ports' )
+        print( '        --disable      Disable all ports' )
+        print( '        --recycle      Recycle all ports' )
+        print( '        --reset        Reset the YKUSH' )
+        sys.exit(2)
+    try:
+        opts,args = getopt.getopt( sys.argv[1:], '',
+            longopts = [ 'help', 'recycle', 'enable', 'disable', 'reset' ])
+    except getopt.GetoptError as err:
+        print( '-F-', err )   # something like "option -a not recognized"
+        usage()
+    if args or not opts:
+        usage()
+    # See the end of the file for all the option handling
+
+# we assume that if pykush is not installed, it is not in use
+try:
+    import pykush
+except ModuleNotFoundError:
+    log.w( 'No pykush library is available!' )
+    raise
+
+
+class NoneFoundError(pykush.YKUSHNotFound):
+    def __init__(self, message=None):
+        super().__init__(self, message or 'no YKUSH module found')
+
+
+class Ykush(usb_relay.usb_relay):
+    _ykush = None
+    NUM_PORTS = 3
+
+    def __init__(self):
+        super().__init__()
+        yk = self.discover()
+        if yk == None:
+            raise NoneFoundError()
+        self._ykush = None
+
+    def discover(self, retries = 0, serial = None, path = None):
+        """
+        Return a YKUSH device. Raise YKUSHNotFound if none found
+        If it was called more than once when there's only one device connected, return it
+        """
+        ykush_dev = None
+        for i in range(retries + 1):
+            try:
+                ykush_dev = pykush.YKUSH(serial=serial, path=path)
+            except pykush.YKUSHNotFound as e:
+                log.w("YKUSH device not found!")
+            except Exception as e:
+                log.w("Unexpected error occurred!", e)
+            finally:
+                if not ykush_dev and i < retries:
+                    time.sleep(1)
+                else:
+                    break
+        return ykush_dev
+
+    def connect(self, reset = False, serial = None, path = None):
+        """
+        Connect to a YKUSH device
+        :param reset: When true, the YKUSH will be reset as part of the connection process
+        """
+        if reset:
+            log.d("resetting YKUSH...")
+            import subprocess
+            subprocess.run("ykushcmd ykush3 --reset".split())
+            time.sleep(5)
+
+        if not self._ykush:
+            self._ykush = self.discover(serial=serial, path=path)
+
+        self._set_connected(self, usb_relay.YKUSH)
```

Comment:
Why is this needed?

Reply:
without _relay and _relay_type in the base class it is not needed, will remove that